Unit testing order (#18456)
dotnet/docs: docs/core/testing/selective-unit-tests.md

---
title: Run selective unit tests
description: How to use a filter expression to run selective unit tests with the dotnet test command in .NET Core.
author: smadala
ms.date: 05/18/2020
zone_pivot_groups: unit-testing-framework-set-one
---

# Run selective unit tests

With the [`dotnet test`](../tools/dotnet-test.md) command in .NET Core, you can use a filter expression to run selective tests. This article demonstrates how to filter which tests are run. The following examples use `dotnet test`. If you're using `vstest.console.exe`, replace `--filter` with `--testcasefilter:`.

## Character escaping

Using filters that include exclamation mark `!` on `*nix` requires escaping since `!` is reserved. For example, this filter
skips all tests if the namespace contains IntegrationTests:

```dotnetcli
dotnet test --filter FullyQualifiedName\!~IntegrationTests
```

> [!IMPORTANT]

> > Notice the backslash that precedes the exclamation mark, that indicates it is an escaped character `\!`.

Reviewer: suggest rewording to avoid *that* repeated:

```suggestion
> The backslash precedes the exclamation mark to indicate it is an escaped character `\!`.
```
